### ISSUE TITLE
[BUG] Dataset default_mask_target or mask_target not used by evaluate_segmentations()

### ISSUE THREAD

**Issue by allenleetc** [voxel51/fiftyone#3213]
### System information

-   **OS Platform and Distribution**: Linux Ubuntu 20.04
-   **Python version** (`python --version`): 3.10
-   **FiftyOne version** (`fiftyone --version`): 0.21.0
-   **FiftyOne installed from** (pip or source): pip

### Commands to reproduce

Code from doc example at https://docs.voxel51.com/user_guide/evaluation.html#id17

```
import fiftyone as fo
import fiftyone.zoo as foz

# Load a few samples from COCO-2017
dataset = foz.load_zoo_dataset(
    "quickstart",
    dataset_name="segmentation-eval-demo",
    max_samples=10,
    shuffle=True,
)

# The models are trained on the VOC classes
CLASSES = (
    "background,aeroplane,bicycle,bird,boat,bottle,bus,car,cat,chair,cow," +
    "diningtable,dog,horse,motorbike,person,pottedplant,sheep,sofa,train," +
    "tvmonitor"
)
dataset.default_mask_targets = {
    idx: label for idx, label in enumerate(CLASSES.split(","))
}

# Add DeepLabv3-ResNet101 predictions to dataset
model = foz.load_zoo_model("deeplabv3-resnet101-coco-torch")
dataset.apply_model(model, "resnet101")

# Add DeepLabv3-ResNet50 predictions to dataset
model = foz.load_zoo_model("deeplabv3-resnet50-coco-torch")
dataset.apply_model(model, "resnet50")

print(dataset)

# Evaluate the masks w/ ResNet50 backbone, treating the masks w/ ResNet101
# backbone as "ground truth"
results = dataset.evaluate_segmentations(
    "resnet50",
    gt_field="resnet101",
    eval_key="eval_simple",
)

# Get a sense for the per-sample variation in likeness
print("Accuracy range: (%f, %f)" % dataset.bounds("eval_simple_accuracy"))
print("Precision range: (%f, %f)" % dataset.bounds("eval_simple_precision"))
print("Recall range: (%f, %f)" % dataset.bounds("eval_simple_recall"))

# Print a classification report
results.print_report()
```

Output does not use map for labels, nor is space of classes restricted if map is restricted to certain classes (as would occur if this map were passed directly as a `mask_targets` kwarg:

```
Recall range: (0.144033, 0.999115)
              precision    recall  f1-score   support

           1       0.95      0.95      0.95    2781.0
           3       0.97      0.99      0.98   62381.0
           6       0.00      0.00      0.00     321.0
           7       0.88      0.39      0.54   19571.0
           8       0.99      1.00      0.99  186358.0
          12       0.99      0.99      0.99  104431.0
          15       0.94      0.78      0.85   29337.0
          17       0.00      0.00      0.00      31.0
          19       0.00      0.00      0.00       0.0

   micro avg       0.97      0.95      0.96  405211.0
   macro avg       0.64      0.57      0.59  405211.0
weighted avg       0.98      0.95      0.96  405211.0
```
### Describe the problem

Probably the behavior of  `evaluate_segmentations` when `default_mask_targets` or `mask_targets` is present should be as if that map were passed directly as a `mask_targets` kwarg.

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

**Comment by brimoor** [comment]
I believe this is a case of misleading documentation that I just fixed [here](https://github.com/voxel51/fiftyone/commit/62e7b69a51e6eae7bece815ada0acbbd057ab0e6).

For context, it used to be the case that `classes` and `mask_targets` were automatically used by default by methods like `export()` and `evaluate_XXX()`, but users found this confusing for a couple reasons, including:
- If they write a view that filters to only include specific classes, they intuitively expect that evaluation results would only include those classes, not all possible classes on the dataset
- Some importers like COCO "secretly" populate `default_classes` without the user necessarily knowing. This lead to confusion when, for example, adding/removing classes later without updating `default_classes` (since they didn't know about it)

tldr I like the current behavior which is [now](https://github.com/voxel51/fiftyone/commit/62e7b69a51e6eae7bece815ada0acbbd057ab0e6) appropriately documented: classes and mask targets can be stored, will be always be used by the App for rendering, and can optionally be accessed and provided manually to methods like `export()` and `evaluate_XXX()`.
